plugin/src/ui/components/markdown/index.tsx

```diff
@@ -10,14 +10,24 @@ interface MarkdownProps {
 const Markdown: React.FC<MarkdownProps> = ({ content, className }) => {
   const renderChild = RenderChildContext.use();
   const ref = useRef<HTMLDivElement>(null);
+  let componentRef = useRef<Component | null>(null);
 
   useEffect(() => {
     const renderMarkdown = async () => {
       if (ref.current === null) {
         return;
       }
 
-      await MarkdownRenderer.renderMarkdown(content, ref.current, "", renderChild);
+      // Create a new component or use existing one
+      if (componentRef.current === null) {
+        componentRef.current = new Component();
+      }
+
+      // Attach renderChild to the component
+      componentRef.current.addChild(renderChild);
```

Comment:
I'm still unsure that this is correct. `renderChild` is a reference to the Obsidian component wrapping the injected React component, so the relationship feels like it should be the other way around, i.e. - `renderChild` is the parent of the new component.

Looking at the docs for `MarkdownRenderer.renderMarkdown`:

```
     * @param component - A parent component to manage the lifecycle of the rendered child components.
```

Why doesn't `renderChild` satisfy this, it is a subclass of `Component` already - https://github.com/jamiebrynes7/obsidian-todoist-plugin/blob/master/plugin/src/query/injector.tsx#L47?

Can you reproduce a situation whereby we can produce this error on the current state on `master`?